app/jobs/ranks_processing_job.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class RanksProcessingJob < ApplicationJob
+  queue_as :default
+
+  def perform(users, realm, platform)
+    user_rows = []
+    rank_rows = []
+
+    user_identities = Identity.where(identifier: users.keys, platform: platform)
+                              .pluck(%i[identifier user_id])
+                              .to_h
```

Comment:
TODO: Change this to how it's done in the callbacks controller to account for multiple user_ids for the same identifier (very rare)